MonstieWash/Assets/Scripts/Consumables/ChildConsumables/Treat.cs

```diff
@@ -16,25 +18,36 @@ public struct MoodEffect
         [Tooltip("Which moods this treat will effect")] public MoodType target;
         [Tooltip("How much the affected moods will alter by, should match values 1-1 with values in moodtarget list")] public float effect;
     }
+
     /// <summary>
     /// Removes one of this consumable from storage and apply its effect to its targeted mood.
     /// </summary>
     public override void Consume()
     {
-        Debug.Log("Finding Brain");
         //Find Monster Brain
         var brain = FindFirstObjectByType<MonsterBrain>();
-        Debug.Log("Found brain");
+
         //Loop through mood targets and update them by the effect amount;
-        for (var i = 0; i < moods.Count; i++) 
+        for (var i = 0; i < moodEffects.Count; i++) 
         {
-            if (!brain.Moods.Contains(moods[i].target)) continue;
-            brain.UpdateMood(moods[i].effect, moods[i].target);
-            Debug.Log("Updating: " + moods[i].target + " by " + moods[i].effect);
+            if (!brain.Moods.Contains(moodEffects[i].target)) continue;
+
+            brain.UpdateMood(moodEffects[i].effect, moodEffects[i].target);
+            if (showDebug) Debug.Log("Updating: " + moodEffects[i].target + " by " + moodEffects[i].effect);
+
             UseTreat?.Invoke();
         }
 
         //Remove one of this consumable type from the manager.
         FindFirstObjectByType<Inventory>().UpdateStorageAmount(this,-1);
+
+        //Play mood FX
+        var moodFXManager = FindFirstObjectByType<MoodFXManager>();
```

Comment:
Nitpick but, could do this on awake/start so it only has to find it once.
(I probably wouldn't have done this either though)

Reply:
Oh yeah this is like this because the bag of treats is in the startingScene and the `MoodFXManager` is in each monster scene (like front, side, back etc.) so it depends which scene you're in for which `MoodFXManager` will be used, so it kinda has to be done here anyways.